addons/rating/models/rating_mixin.py

```diff
@@ -42,12 +63,27 @@ def _compute_rating_percentage_satisfaction(self):
                 grades_per_parent[parent_id]['okay'] += item['__count']
             else:
                 grades_per_parent[parent_id]['bad'] += item['__count']
+            rating_scores_per_parent[parent_id] += rating * item['__count']
 
         # compute percentage per parent
         for record in self:
             repartition = grades_per_parent.get(record.id, default_grades)
-            record.rating_count = sum(repartition.values())
-            record.rating_percentage_satisfaction = repartition['great'] * 100 / sum(repartition.values()) if sum(repartition.values()) else -1
+            rating_count = sum(repartition.values())
+            record.rating_count = rating_count
+            record.rating_percentage_satisfaction = repartition['great'] * 100 / rating_count if rating_count else -1
+            record.rating_avg = rating_scores_per_parent[record.id] / rating_count if rating_count else 0
+            record.rating_avg_percentage = record.rating_avg / 5
+
+    def _search_rating_avg(self, operator, value):
+        if operator not in OPERATOR_MAPPING:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        domain = [('parent_res_model', '=', self._name), ('consumed', '=', True), ('rating', '>=', RATING_LIMIT_MIN)]
+        if self._rating_satisfaction_days:
+            min_date = fields.datetime.now() - timedelta(days=self._rating_satisfaction_days)
+            domain = expression.AND([domain, [('write_date', '>=', fields.Datetime.to_string(min_date))]])
+        rating_read_group = self.env['rating.rating'].sudo().read_group(domain, ['parent_res_id', 'rating_avg:avg(rating)'], ['parent_res_id'])
+        parent_res_ids = [res['parent_res_id'] for res in rating_read_group if OPERATOR_MAPPING[operator](float_compare(res['rating_avg'], value, 2), 0)]
```

Comment:
Do you prefer writing that on many lines? 🤔 
Like this:
```suggestion
        parent_res_ids = [
            res['parent_res_id']
            for res in rating_read_group
            if OPERATOR_MAPPING[operator](float_compare(res['rating_avg'], value, 2), 0)
        ]
```